services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_service_specs/_constarts.py

```diff
@@ -0,0 +1,6 @@
+from typing import Final
+
+_NANO: Final[int] = int(1e9)
+
+SERVICE_RESTART_MAX_ATTEMPTS: Final[int] = 100
+SERVICE_RESTART_DELAY_BETWEEN_RESTARTS_S: Final[int] = 1 * _NANO
```

Comment:
so if a service fails, it shall restart in 1ns and a maximum of 100 times?
hmm, not sure I get the rationale here... why define a time at all if it's so small?

Reply:
just wondering, why in that case you do not put just 'always' for restart policy, and keep the default of 5s...

Reply:
Are you sure the default was 5 seconds? From what I got there it was 5 ms. From what I understood the unit of measurement is in nanoseconds. Meaning that this was set to 5 milliseconds.
Would you know a much reliable source where to check for this value? In the docker API I could not find anything.

Reply:
look here for docker compose defaults: https://docs.docker.com/compose/compose-file/compose-file-v3/
look here for docker REST API defaults: https://docs.docker.com/engine/api/v1.41/#tag/Service/operation/ServiceCreate
look here for docker cli defaults: https://docs.docker.com/engine/reference/commandline/service_create/

REST API is probably using the exact same defaults than the cli does. which is delay of 5seconds if I read correctly

Reply:
I also went through all these and the delay is 5 seconds OK. But the bloody API docs do not specify the unit they use. They only state it is `int64`. Putting together the information coming from the other docs, it makes me believe the unit is nanoseconds in the API.
Thoughts?

Reply:
if 5s is ok, I would not bother changing it.